hub/core/transform/transform.py

```diff
@@ -0,0 +1,164 @@
+from hub.util.exceptions import (
+    InvalidInputDataError,
+    TensorMismatchError,
+    UnsupportedSchedulerError,
+)
+from hub.util.transform import (
+    pad_or_shrink_kwargs,
+    load_updated_meta,
+    merge_index_metas,
+    merge_tensor_metas,
+    transform_sample,
+)
+from typing import Callable, Dict, List, Optional, Sequence, Set, Tuple
+from hub.util.remove_cache import get_base_storage
+from hub.core.storage import LRUCache, MemoryProvider, StorageProvider
+from hub.core.compute import ThreadProvider, ProcessProvider, ComputeProvider
+from hub.core.meta.tensor_meta import TensorMeta
+from hub.core.meta.index_meta import IndexMeta
+from hub.core.tensor import append_tensor
+from hub.constants import MB
+from itertools import repeat
+from hub import Dataset
+import math
+
+
+def transform(
+    data_in,
+    pipeline: Sequence[Callable],
+    ds_out: Dataset,
+    pipeline_kwargs: Optional[Sequence[Dict]] = None,
+    scheduler: str = "threaded",
+    workers: int = 1,
+):
+
+    """Initializes a new or existing dataset.
```

Comment:
Docstring? Did this get copy/pasted from `Dataset.__init__`? This is a really important API to have well-documented, I think we can add at least a few more lines here plus some examples.